src/lib.rs

```diff
@@ -1674,6 +1674,77 @@ impl<B: BitBlock> BitVec<B> {
             carry = curr_carry;
         }
     }
+
+    // todo(jujumba): docs
+    pub fn insert_many(&mut self, at: usize, bits: &[bool]) {
+        assert!(
+            at <= self.nbits,
+            "insertion index (is {at}) should be <= nbits (is {nbits})",
+            nbits = self.nbits
+        );
+
+        let block_at = at / B::bits();
+        let bit_at = at % B::bits();
+        let block_offset = bits.len() / B::bits();
+
+        self.extend_if_needed(self.nbits + bits.len());
+
+        unsafe {
+            self.rotate(at, bits.len() % B::bits());
+        }
+
+        for i in ((block_at + 1)..=(block_at + block_offset)).rev() {
+            let block = core::mem::replace(&mut self.storage[i], B::zero());
+            self.storage[i + block_offset] = block;
+        }
+
+        let mask = (B::one() << bit_at) - B::one();
+        let carry = self.storage[block_at] & !mask;
+        self.storage[block_at] = self.storage[block_at] & mask;
+        self.storage[block_at + block_offset] = self.storage[block_at + block_offset] | carry;
+
+        self.nbits += bits.len();
+
+        for (index, bit) in bits
+            .iter()
+            .copied()
+            .map(|bit| if bit { B::one() } else { B::zero() })
+            .enumerate()
+        {
+            self.storage[(at + index) / B::bits()] = self.storage[(at + index) / B::bits()] | bit;
```

Comment:
Moving bit-by-bit is slow, in fact, 32 times slower.

Reply:
We should eventually accept BitSlice right here, once that is done.

Reply:
> Moving bit-by-bit is slow, in fact, 32 times slower.

There is no other way to do it, at least now

Reply:
@Jujumba Exposing it in the public API and docs makes us kinda commit to this style of taking in bits. We should probably wait until I have enough time on my hands to do a BitSlice.

Reply:
> @Jujumba Exposing it in the public API and docs makes us kinda commit to this style of taking in bits. We should probably wait until I have enough time on my hands to do a BitSlice.

Okay, so should I close this PR?

And could you, please, review another PR I submitted (when you have some spare time, indeed) :pleading_face:

Reply:
Sure.

I think best for this is `bits: impl Iterator<Item = bool>` and grab 64 bits at a time